ducktape/cluster/remoteaccount.py

```diff
@@ -142,97 +317,182 @@ def kill_process(self, process_grep_str, clean_shutdown=True, allow_fail=False):
+            #   path/to/src_name -> dest/src_name
+            src_name = src
+            if src_name.endswith(os.path.sep):
+                src_name = src_name[:-len(os.path.sep)]  # trim off path separator from end
+            src_name = os.path.basename(src_name)
+
+            dest = os.path.join(dest, src_name)
+
+        if self.isfile(src):
+            self.sftp_client.get(src, dest)
+        elif self.isdir(src):
+            # we can now assume dest path looks like: path_that_exists/new_directory
+            os.mkdir(dest)
+
+            # for obj in `ls src`, if it's a file, copy with copy_file_from, elif its a directory, call again
+            for obj in self.sftp_client.listdir(src):
+                obj_path = os.path.join(src, obj)
+                if self.isfile(obj_path) or self.isdir(obj_path):
+                    self.copy_from(obj_path, dest)
+                else:
+                    # TODO what about uncopyable file types?
+                    pass
+
+    def copy_to(self, src, dest):
+
+        if self.isdir(dest):
+            # dest is an existing directory, so assuming src looks like path/to/src_name,
+            # in this case we'll copy as:
+            #   path/to/src_name -> dest/src_name
+            src_name = src
+            if src_name.endswith(os.path.sep):
+                src_name = src_name[:-len(os.path.sep)]  # trim off path separator from end
+            src_name = os.path.basename(src_name)
+
+            dest = os.path.join(dest, src_name)
+
+        if os.path.isfile(src):
+            # local to remote
+            self.sftp_client.put(src, dest)
+        elif os.path.isdir(src):
+            # we can now assume dest path looks like: path_that_exists/new_directory
+            self.mkdir(dest)
+
+            # for obj in `ls src`, if it's a file, copy with copy_file_from, elif its a directory, call again
+            for obj in os.listdir(src):
+                obj_path = os.path.join(src, obj)
+                if os.path.isfile(obj_path) or os.path.isdir(obj_path):
+                    self.copy_to(obj_path, dest)
+                else:
+                    # TODO what about uncopyable file types?
+                    pass
 
-    def rsync_to_command(self, flags, src_dir, dest_dir):
-        r = "rsync "
-        if self.ssh_args:
-            r += "-e \"ssh " + self.ssh_args + "\" "
-        if flags:
-            r += flags
-        r += src_dir
-        if self.user:
-            r += self.user + "@"
-        r += self.hostname + ":" + dest_dir
-        return r
+    def islink(self, path):
+        try:
+            # stat should follow symlinks
+            path_stat = self.sftp_client.lstat(path)
+            return stat.S_ISLNK(path_stat.st_mode)
+        except:
+            # TODO figure out which errors are legit (e.g. if file does not exist, what will sftp.stat do?)
+            is_file = False
 
-    def rsync_to(self, flags, src_dir, dest_dir):
```

Comment:
Yeah I agree this is a lot faster in the right situation. This is mostly practical - there aren't currently known uses, and I think it will take a notrivial amount of time and experimentation to get it to work w/paramiko.